observation_portal/blocks/conversion.py

```diff
@@ -0,0 +1,478 @@
+import copy
+from observation_portal.requestgroups.models import RequestGroup
+
+
+POND_POINTING_TYPE_TO_TARGET_TYPE = {
+    'SP': 'SIDEREAL',
+    'NS': 'NON_SIDEREAL',
+    'ST': 'SATELLITE',
+}
+
+TARGET_TYPE_TO_POND_POINTING_TYPE = {
+    'SIDEREAL': 'SP',
+    'NON_SIDEREAL': 'NS',
+    'SATELLITE': 'ST',
+    'STATIC': 'ST',
+}
+
+BLOCK_REQUIRED_FIELDS = ['instrument_class', 'site', 'observatory', 'telescope', 'start', 'end']
+MOLECULE_REQUIRED_FIELDS = ['type', 'inst_name']
+
+
+class PondBlockError(Exception):
+    pass
+
+
+def convert_pond_blocks_to_observations(blocks):
+    if isinstance(blocks, list):
+        return [convert_pond_block_to_observation(block) for block in blocks]
+    else:
+        return convert_pond_block_to_observation(blocks)
+
+
+def convert_pond_block_to_observation(block):
+    if not block.get('molecules', []):
+        raise PondBlockError("blocks must contain at least one molecule")
+
+    for field in BLOCK_REQUIRED_FIELDS:
+        if field not in block:
+            raise PondBlockError("Blocks must include {} field".format(field))
+
+    first_molecule = block['molecules'][0]
+    observation = {
+        'site': block.get('site', ''),
+        'enclosure': block.get('observatory', ''),
+        'telescope': block.get('telescope', ''),
+        'start': block.get('start', ''),
+        'end': block.get('end', ''),
+        'name': first_molecule.get('group', ''),
+        'proposal': first_molecule.get('prop_id', '')
+    }
+
+    if first_molecule.get('user_id'):
```

Comment:
yes... this will fail validation so the person submitting will have to remove user_id so it gets automatically set as the shim submitter user. I'm not sure how to get around this, because user links to an actual user in the DB, so they can't just submit a non-existent one. I guess we can just not attempt to set it at all?

Reply:
oh actually I think no one submits with user_id - so this code is just here to convert existing observations correctly to blocks and back, similar to the state code below it. I was trying to get them as close as possible to verify it worked by converting back and forth between observations and blocks.

Reply:
we decided to leave it as is for now since it shouldn't affect users submitting through the shim as that should overwrite the field.